src/v/cluster/partition.cc

```diff
@@ -60,13 +61,40 @@ ss::future<result<raft::replicate_result>> partition::replicate(
     return _raft->replicate(std::move(r), opts);
 }
 
+raft::replicate_stages partition::replicate_in_stages(
+  model::record_batch_reader&& r, raft::replicate_options opts) {
+    return _raft->replicate_in_stages(std::move(r), opts);
+}
+
 ss::future<result<raft::replicate_result>> partition::replicate(
   model::term_id term,
   model::record_batch_reader&& r,
   raft::replicate_options opts) {
     return _raft->replicate(term, std::move(r), opts);
 }
 
+raft::replicate_stages partition::replicate_in_stages(
```

Comment:
I don't see we use replicate_in_stages in this PR, is this needed for the future changes?

Reply:
exactly, I have next step ready, will post it soon